apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java

```diff
@@ -1916,6 +1927,90 @@ private static CareTeamComponent addCareTeamPractitioner(
+   * @param practitionerIdSystem the {@link Identifier#getSystem()} of the practitioner to reference
+   *     in {@link CareTeamComponent#getProvider()}
+   * @param practitionerIdValue the {@link Identifier#getValue()} of the practitioner to reference
+   *     in {@link CareTeamComponent#getProvider()}
+   * @param careTeamRole the {@link ClaimCareteamrole} to use for the {@link
+   *     CareTeamComponent#getRole()}
+   * @return the {@link CareTeamComponent} that was created/linked
+   */
+  public static CareTeamComponent addCareTeamPerforming(
+      ExplanationOfBenefit eob,
+      ItemComponent eobItem,
+      C4BBClaimProfessionalAndNonClinicianCareTeamRole role,
+      Optional<String> npiValue,
+      Optional<String> upinValue,
+      Optional<String> pinValue,
+      boolean includeTaxNumbers,
+      String taxValue) {
+
+    List<Identifier> identifiers = new ArrayList<Identifier>();
+
+    if (npiValue.isPresent()) {
+      identifiers.add(
+          TransformerUtilsV2.createC4BBPractionerIdentifier(
+              C4BBPractitionerIdentifierType.NPI, npiValue.get()));
+    }
+
+    if (upinValue.isPresent()) {
+      identifiers.add(
+          TransformerUtilsV2.createC4BBPractionerIdentifier(
+              C4BBPractitionerIdentifierType.UPIN, upinValue.get()));
+    }
+
+    if (pinValue.isPresent()) {
+      identifiers.add(
+          TransformerUtilsV2.createC4BBPractionerIdentifier(
+              C4BBPractitionerIdentifierType.PIN, pinValue.get()));
+    }
+
+    if (includeTaxNumbers) {
+      identifiers.add(
+          TransformerUtilsV2.createC4BBPractionerIdentifier(
+              C4BBPractitionerIdentifierType.TAX, taxValue));
+    }
+
+    return addCareTeamPractitionerForPerforming(eob, eobItem, role, identifiers);
+  }
+
+  public static CareTeamComponent addCareTeamPractitionerForPerforming(
+      ExplanationOfBenefit eob,
+      ItemComponent eobItem,
+      C4BBClaimProfessionalAndNonClinicianCareTeamRole role,
+      List<Identifier> identifiers) {
+    // Try to find a matching pre-existing entry.
+    CareTeamComponent careTeamEntry = eob.addCareTeam();
+    // addItem adds and returns, so we want size() not size() + 1 here
+    careTeamEntry.setSequence(eob.getCareTeam().size());
+
+    Practitioner practitioner = createContainedPractitioner(eob);
+    practitioner.setIdentifier(identifiers);
+
+    Reference ref = new Reference(practitioner);
+
+    careTeamEntry.setProvider(ref);
+
+    CodeableConcept careTeamRoleConcept = createCodeableConcept(role.getSystem(), role.toCode());
+    careTeamRoleConcept.getCodingFirstRep().setDisplay(role.getDisplay());
+    careTeamEntry.setRole(careTeamRoleConcept);
+
+    // ExplanationOfBenefit.careTeam.sequence => ExplanationOfBenefit.item.careTeamSequence
+    if (!eobItem.getCareTeamSequence().contains(new PositiveIntType(careTeamEntry.getSequence()))) {
+      eobItem.addCareTeamSequence(careTeamEntry.getSequence());
+    }
+
+    return careTeamEntry;
+  }
```

Comment:
This is great!  I like having this all organized here to connect the provider in the contains with the reference.